synapse/lib/auth.py

```diff
@@ -81,3 +97,900 @@ def runas(user):
+        mname, mdict = mesg
+        name = mdict.get('user')
+        rule = mdict.get('rule')
+        uobj = self.auth.reqUser(name)
+        uobj.addRule(rule)
+        ret = s_tufo.tufo(mname, user=(name, uobj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authDelUserRule(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('user')
+        rule = mdict.get('rule')
+        uobj = self.auth.reqUser(name)
+        uobj.delRule(rule)
+        ret = s_tufo.tufo(mname, user=(name, uobj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authAddRoleRule(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('role')
+        rule = mdict.get('rule')
+        robj = self.auth.reqRole(name)
+        robj.addRule(rule)
+        ret = s_tufo.tufo(mname, role=(name, robj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authDelRoleRule(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('role')
+        rule = mdict.get('rule')
+        robj = self.auth.reqRole(name)
+        robj.delRule(rule)
+        ret = s_tufo.tufo(mname, role=(name, robj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authAddAdmin(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('user')
+        uobj = self.auth.reqUser(name)
+        uobj.setAdmin(True)
+        ret = s_tufo.tufo(mname, user=(name, uobj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authDelAdmin(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('user')
+        uobj = self.auth.reqUser(name)
+        uobj.setAdmin(False)
+        ret = s_tufo.tufo(mname, user=(name, uobj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authAddUserRole(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('user')
+        role = mdict.get('role')
+        uobj = self.auth.reqUser(name)
+        robj = self.auth.reqRole(role)
+        uobj.addRole(role)
+        ret = s_tufo.tufo(mname, user=(name, uobj._getAuthData()))
+        return True, ret
+
+    @reqAdmin
+    def __authDelUserRole(self, mesg):
+        mname, mdict = mesg
+        name = mdict.get('user')
+        role = mdict.get('role')
+        uobj = self.auth.reqUser(name)
+        robj = self.auth.reqRole(role)
+        uobj.delRole(role)
+        ret = s_tufo.tufo(mname, user=(name, uobj._getAuthData()))
+        return True, ret
+
+class Auth(s_config.Config):
+    '''
+    An authorization object which can help enforce cortex rules.
+
+    Args:
+        dirn (str): Dictionry backing the Auth data.
+        conf (dict): Optional configuration data.
+    '''
+    def __init__(self, dirn, conf=None):
+
+        s_config.Config.__init__(self, opts=conf)
+
+        path = os.path.join(dirn, 'auth.lmdb')
+
+        mapsize = self.getConfOpt('lmdb:mapsize')
+
+        self.lenv = lmdb.open(path, max_dbs=128)
```

Comment:
Could just specify `map_size` as a kwarg in the call to `open`. @jnwatson - would that be problematic?

Reply:
totally missed this. good find.

Reply:
holding this till master merge

Reply:
ok